### ISSUE TITLE
upstream bug: Intelephense 1.3.5+ hangs for 2 minutes when activating a license behind proxy

### ISSUE THREAD

**Issue by cprn** [marlonfan/coc-phpls#42]
**Describe the bug**
On initialisation Intelephense fails after 2 minutes when trying to read the workspace state, even though at first it's empty. After timing out it properly re-indexes the project and saves its state. On next open it hangs again and this time doesn't write (probably because there are no changes). I just need a confirmation this isn't `vim`/`coc-phpls` related before I open an issue upstream.

**To Reproduce**
1. Install `vim`
2. Install `coc.nvim`
3. Install `coc-phpls`
4. Open any PHP project

**Expected behavior**
I expect Intelephense to load quickly, not hang while reading state and then timeout.

**Screenshots**
Related parts of `:CocInfo` output:
```md
## versions
vim version: NVIM v0.3.8
node version: v12.13.0
coc.nvim version: 0.0.74-bc017b02e6
term: screen-256color
platform: linux

## Output channel: phpls
[Info  - 15:09:55] Initialising intelephense 1.3.5
[Info  - 15:09:55] Reading state from /home/user/.config/coc/extensions/coc-phpls-data/674614e2.
[Warn  - 15:12:06] undefined
[Info  - 15:12:06] Initialised in 130624 ms
[Info  - 15:12:06] Intelephense 1.2.3
[Info  - 15:12:06] Indexing started.
[Info  - 15:12:07] Indexing ended. 98 files indexed in 1s.
[Info  - 15:12:07] Indexing started.
[Info  - 15:12:22] Indexing ended. 10169 files indexed in 14s.
[Info  - 15:12:22] Writing state to /home/user/.config/coc/extensions/coc-phpls-data/674614e2.
[Info  - 15:12:23] Wrote state in 1.7s.
```

`coc-settings.json`:
```json
{   
    "intelephense.licenceKey": "***",
    "intelephense.files.exclude": ["var"]
}
```

**Desktop (please complete the following information):**
 - OS: Linux Ubuntu 18.04.3 4.15.0-72-generic
 - Version: `coc-phpls`: 2.1.4, `Intelephense`: 1.3.5

**Comment by marlonfan** [comment]
@cprn 
Can you turn on verbose so I can see CocInfo?

**Comment by cprn** [comment]
@marlonfan 
* [1st run](https://gist.github.com/cprn/309316c39bb2b70e42e01f1af2b2e379/99993a0fda2387f78067bce68833139e5ad76c58)
* [2nd run](https://gist.github.com/cprn/309316c39bb2b70e42e01f1af2b2e379/6a7369c4912e8810d25cf2265152e4687a74ca16)
* [diff](https://gist.github.com/cprn/309316c39bb2b70e42e01f1af2b2e379/revisions#diff-7c149053b911ddbb52eca335f1c0db00)

Licence activation issue is due to proxy, I don't want to bother admin each time I reinstall from scratch.

**Comment by cprn** [comment]
"Reading state from" happens in `node_modules/intelephense/lib/intelephense.js` so I'm posting this upstream.

**Comment by cprn** [comment]
OK, so I was testing this some more to open a concise issue upstream and I found out this only happens when I have `intelephense.licenceKey` set in my `coc-settings.json`. It might only be when I'm behind proxy but can't confirm - I currently have no environment that isn't behind proxy.

**Comment by cprn** [comment]
I confirmed with Wireshark this is caused by Intelephense trying to connect to the license server directly (i.e. not respecting environment proxy settings). Ben stated in [this comment](https://github.com/bmewburn/vscode-intelephense/issues/754#issuecomment-565844844) the proxy setting is taken from `process.env.https_proxy` setting but there's no such thing in `coc-settings.json` schema.

**Comment by cprn** [comment]
Can be closed, fixed upstream: bmewburn/vscode-intelephense#915.
